sources/packages/api/src/database/entities/student-assessment.model.ts

```diff
@@ -0,0 +1,208 @@
+   */
+  @Column({
+    name: "trigger_type",
+    type: "enum",
+    enum: AssessmentTriggerType,
+    enumName: "AssessmentTriggerType",
+    nullable: false,
+  })
+  triggerType: AssessmentTriggerType;
+  /**
+   * Offering id that must be used for any assessment/reassessment. This information can
+   * be null only during a PIR process. Upon a program/offering change, this will also
+   * represent the new/changed program/offering.
+   */
+  @RelationId(
+    (studentAssessment: StudentAssessment) => studentAssessment.offering,
+  )
+  offeringId?: number;
+  /**
+   * Offering that must be used for any assessment/reassessment. This information can
+   * be null only during a PIR process. Upon a program/offering change, this will also
+   * represent the new/changed program/offering.
+   */
+  @ManyToOne(() => EducationProgramOffering, {
+    eager: false,
+    cascade: false,
+    nullable: true,
+  })
+  @JoinColumn({
+    name: "offering_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  offering?: EducationProgramOffering;
+  /**
+   * When the reassessment happen due to a student appeal, this will provide to
+   * the workflow the data that need to be changed.
+   */
+  @RelationId(
+    (studentAssessment: StudentAssessment) => studentAssessment.studentAppeal,
+  )
+  studentAppealId?: number;
+  /**
+   * When the reassessment happen due to a student appeal, this will provide to
+   * the workflow the data that need to be changed.
+   */
+  @ManyToOne(() => StudentAppeal, {
+    eager: false,
+    cascade: true,
+    nullable: true,
+  })
+  @JoinColumn({
+    name: "student_appeal_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  studentAppeal?: StudentAppeal;
+  /**
+   * When the reassessment happen due to a scholastic standing change (e.g. student withdrawal),
+   * this will provide to the workflow the data that need be changed.
+   */
+  @ManyToOne(() => StudentScholasticStanding, {
+    eager: false,
+    cascade: true,
+    nullable: true,
+  })
+  @JoinColumn({
+    name: "student_scholastic_standing_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  studentScholasticStanding?: StudentScholasticStanding;
+  /**
+   * Indicates the status of the NOA approval when the student must approve the
+   * money values prior to the institution COE approval and disbursements or when
+   * a reassessment happened and a new approval is required.
+   */
+  @Column({
+    name: "noa_approval_status",
+    type: "enum",
+    enum: AssessmentStatus,
+    enumName: "AssessmentStatus",
+  })
+  noaApprovalStatus: AssessmentStatus;
+}
+
+/**
+ * Interface for BaseAssessment values that are shared between FullTime and PartTime
+ */
+interface BaseAssessment {
+  weeks: number;
+  tuitionCost: number;
+  childcareCost: number;
+  transportationCost: number;
+  booksAndSuppliesCost: number;
+  totalFederalAward: number;
+  totalProvincialAward: number;
+  totalFamilyIncome: number;
+  totalAssessmentNeed: number;
+}
```

Comment:
Can this interface and other interfaces that extend it be moved to separate files ? Also does it not need to have other attributes like createdAt, updateAt ?

Reply:
Regarding this: "_Can this interface and other interfaces that extend it be moved to separate files ?_", yes they can. These interfaces are not new and there were kept here to follow the same way that was already in place.

Regarding this: "Also does it not need to have other attributes like createdAt, updatedAt ?", _these_ interfaces _represents_ the possible content of a _jsonB_ column, not an entitity.